File: leetcode94.java
/**
 * 二叉树的中序遍历
 */

 class Solution{
    public List<Integer> inorderTraversal(TreeNode root){
        List<Integer> ans = new ArrayList<>();
        inorder(ans,root);
        return ans;
    }
    public void inorder(List<Integer> ans,TreeNode root){
        if(root == null){
            return;
        }
        inorder(ans,root.left);
        ans.add(root.val);
        inorder(ans,root.right);
    }
 }

File: leetcodeJAVA/leetcode94.js
/**
 * 二叉树中序遍历
 * @param {*} root 
 * @returns 
 */

let inorderTraversal = function (root){
    const ret = []
    const inorder = (node,ret)=>{
        if(!root){
            return
        }
        inorder(node.left,ret);
        ret.push(root.val);
        inorder(node.right,ret);
    }
    inorder(root);
    return ret;
}